Salesforce: Update templates to steps with fields

anhdungpham opened the pull request:
#### PR Review Checklist
Update templates to steps with fields and removing the Transform Mapping steps.

Readme
- [ ] Are the install steps clear
- [ ] Do all of the links work

mesa.json
- [ ] Key: does it reflect the folder structure? For instance, if the folder for the template is `shopify/order/send_to_another_system`, the key should be the same. 
- [ ] Name: is it logical, proper-cased?
- [ ] Description: is it logical, end in period?
- [ ] Tags: Do they exist on getmesa.com/templates, proper-cased, logical?
- [ ] Source: lowercase
- [ ] Destination: lowercase
- [ ] Enabled: If no configuration necessary `true`, if there are ANY setup steps `false`
- [ ] Do the Input/Output names make sense? How about the keys?
- [ ] Are Storage, Secrets and scripts well-named

Template code
- [ ] Is code readable and well-commented?

QA
- [ ] Send Customer to Salesforce Contact
- [ ] Send Order to Salesforce Contact
- [ ] Send Order to Salesforce Lead
- [ ] Send Salesforce Product To Shopify Product
- [ ] Send Salesforce Lead To Shopify Customer
- [ ] Does the template work


#### Deploy checklist
- [ ] Squash and merge PR
- [ ] Add template key to https://homeroom.theshoppad.com/admin/#/mesa-templates

amandahom commented:
Collab: https://partners.shopify.com/111585/stores/1943863394

**Send Salesforce Lead To Shopify Customer:**
- The customer gets created but I dont see any relevant information on the customer.
Seems like the entire Address field and email is missing:
![Screen Shot 2022-05-13 at 1 48 04 PM](https://user-images.githubusercontent.com/42184178/168387936-b12979da-d441-455e-ab3c-aff19ed82622.png)
Task: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-amanda/mesa/automations/627eaf26b7a6444d9d50b1e7/queue/task/627ec3e478b7ff04fa1b20b4
Customer: https://shoppad-amanda.myshopify.com/admin/customers/6231089348706

anhdungpham commented:
@amandahom Thanks for taking a look!

**Send Salesforce Lead To Shopify Customer**
- Good catch! There is an issue with the "Shopify Create Customer" step and setting up the Addresses piece so it will transfer over properly to the Shopify Customer. I created an Asana ticket: https://app.asana.com/0/393037162945950/1202261869487287
- Please disregard this one for now. I reverted the changes back for this template.

**Send Customer to Salesforce Contact**
-  I added the Address 2 token.

**Send Order to Salesforce Contact** 
- The customer "Winnie The Pooh" was already a contact in Salesforce. Once I removed the contact and replayed your workflow, it ran successfully. Salesforce seems pretty strict with duplicates.
- I added the Address 2 token.

**Send Order to Salesforce Lead** 
- The customer "Blahhhhhhh Blob" was already a contact in Salesforce. Once I removed the contact and replayed your workflow, it ran successfully.Salesforce seems pretty strict with duplicates.
- I added the Address 2 token.

For the last 2, could you please pull in the new changes and test once again? I removed the customers from the Contacts and Leads page on Salesforce.

amandahom commented:
@anhdungpham **Send Order to Salesforce Lead** worked for me! But, I cant get **Send Order to Salesforce Contact** to work 😞

I've been placing test orders with new and random contacts so i dont understand how it can be a duplicate. 

Task: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-amanda/mesa/automations/627ee60f3435b0504b7a70ae/queue/task/627ee745f35dee0df60ae638

anhdungpham commented:
@amandahom It looks like it may be partially matching some of the data to another contact? Not entirely sure.

I copied the payload and adjusted all the fields so it acts like a brand new contact. These are the changes I made:
![Screen Shot 2022-05-13 at 4 33 05 PM](https://user-images.githubusercontent.com/22602497/168401377-ec5beeb9-082b-4ac2-bee5-96db31298fa4.png)
![Screen Shot 2022-05-13 at 4 33 19 PM](https://user-images.githubusercontent.com/22602497/168401383-fb8e3e1f-4985-4345-a876-493d57321b24.png)

I was able to get it to run successfully. Here is my latest run on your workflow: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-amanda/mesa/automations/627ee60f3435b0504b7a70ae/queue/task/627ee97ba798517409072690

Could you please try with completely new information and a different email like priority@theshoppad.com?

amandahom commented:
@anhdungpham I dont know what im doing wrong. I have used completely brand new info here: https://www.theshoppad.com/homeroom.theshoppad.com/admin/backdoor/shoppad-amanda/mesa/automations/627ee60f3435b0504b7a70ae/queue/task/627eedd8a8576e57d54da1d4

Maybe a new set of eyes from next week will help. Or maybe someone has better luck? Sorry for the back and forth!

anhdungpham commented:
@amandahom No worries! Had to figure this out one before the end of the day so it isn't stuck in my head. It looks like the duplicates also extends to Leads. I guess Salesforce considers them the same since you can convert Leads to Customers.

I think I remove the ones we were testing with in both Leads and Contacts. Could you please try once more with that template.

amandahom commented:
@anhdungpham Finally worked (with the order lead workflow is disabled)!! 🥳. have a great weekend.
